src/rpc/common/Validators.hpp

```diff
@@ -530,4 +527,38 @@ struct CustomValidators final {
     static CustomValidator CurrencyIssueValidator;
 };
 
+/**
+ * @brief Validates that the elements of the array is of type Hex256 uint
+ */
+struct Hex256ItemType final {
+    /**
+     * @brief Validates given the prerequisite that the type of the json value is an array,
+     * verifies all values within the array is of uint256 hash
+     *
+     * @param value the value to verify
+     * @param key The key used to retrieve the tested value from the outer object
+     * @return `RippledError::rpcINVALID_PARAMS` if validation failed; otherwise no error is returned
+     */
+    [[nodiscard]] static MaybeError
+    verify(boost::json::value const& value, std::string_view key)
+    {
+        if (not value.is_object() or not value.as_object().contains(key.data()))
+            return {};  // ignore. If field is supposed to exist, let 'required' fail instead
+
+        auto const& res = value.as_object().at(key.data());
+        if (!res.is_array()) {
+            return Error{Status{RippledError::rpcINVALID_PARAMS, "Credentials an array of CredentialID(hash256)"}};
```

Comment:
We'd better not to mention `Credentials` here. It is supposed to be a generic validator for array of uint256.

Reply:
Please add unittest.